### PR TITLE
verify_local_bear do not need to pre-split strings

### PR DESCRIPTION
**Pull request by Ana06**
Closes https://github.com/coala-analyzer/coala-bears/issues/322 :koala:

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
Hello, thats surely a long PR. First of all, your commit message needs some work, check http://coala.readthedocs.io/en/latest/Getting_Involved/Writing_Good_Commits.html on how to write amazing commits. secondly, you have some coding style issues in your code, gitmate points them out. to simply fix them all at once, run ``coala`` in your command line and it will fix them all for you. to install coala, check https://github.com/coala-analyzer/coala#installation

**Comment by Adrianzatreanu** [comment]
Long story short, commit shortlog should be something like ``bears: Strings are not pre-splitted``

**Comment by Makman2** [comment]
about the issues on the CI :)
- *TailorBearTest* L25: There's the dot too much behind `"""`
- You forgot to adapt *ClangBearIgnoreTest*, *ClangBearTest*, *CMakeLintBearConfigTest*, *CMakeLintBearTest*, 
- *ESLintBearTest* L27: This is a file already splitted by hand. Just remove the tuple-braces
- *RubySmellBearTest* L10: The same
- *XMLBearTest*: Starting at L31, the test files are loaded from file and each line is collected via `list`. I would suggest you implement a little function `load_testdata` that does this:
  ```
  def load_testdata(filename):
      path = os.path.join(os.path.dirname(__file__), "test_files", filename)
      with open(path):
          return path.read()
  ```
  and use this function everywhere :+1:
- *GoImportsBearTest*: A line is too long, GitMate left a comment at the exact location :) Just wrap the string :+1:

If you need help with each of these issues, you can ping me here or on the gitter channel :+1: :D

**Comment by Adrianzatreanu** [comment]
hey @Ana06 you still working on this :) ?

**Comment by Ana06** [comment]
@Adrianzatreanu I've been busy, but I'll finish it this weekend :wink:

**Comment by Adrianzatreanu** [comment]
Amazing, thanks for the reply!

**Comment by Asnelchristian** [comment]
Hey @Ana06 i think the good file for GoImportTest should look like:
```
good_file = """'package main
'', 'import "os"', '',
+    (good_file,),
-      'func main() {', '\tf, _ := os.Open("foo")', '}'

File: tests/go/GoImportsBearTest.py [hunk 1]
```diff
@@ -2,9 +2,26 @@
...
+good_file = """
+print("Hello World!")package main
+
+import "os"
+
+func main() {
+\tf, _ := os.Open("foo")
+}
+"""
+
```
Comment (Asnelchristian) [inline tests/go/GoImportsBearTest.py]: i don't think this how this should look like 
```
good_file="""
package main
import os
func main() {
\tf, _ := os.Open("foo")
}"""
:)

File: tests/xml2/XMLBearTest.py [hunk 2]
```diff
@@ -4,54 +4,42 @@
...
+
+valid_xml_path = load_testdata("note.xml")
+
+valid_xml_url = load_testdata("concept-valid.xml")
+
+invalid_xml_schema = load_testdata("xsd-error.xml")
+
+invalid_xml_dtd = load_testdata("dtd-error.xml")
+
+invalid_xml_url = load_testdata("concept-invalid.xml)
```
Comment (Asnelchristian) [inline tests/xml2/XMLBearTest.py]: you forgot the closing ``"``

**Comment by Asnelchristian** [comment]
https://github.com/coala-analyzer/coala-bears/pull/548/commits/09f3e8d09bffb6a52b1c67f8cb44296866550ea6 needs work

**Comment by Asnelchristian** [comment]
you made a typo in your commit message ``splited`` not ``slitted``

File: tests/go/GoImportsBearTest.py [hunk 3]
```diff
@@ -2,9 +2,21 @@
 from bears.go.GoImportsBear import GoImportsBear
 from tests.LocalBearTestHelper import verify_local_bear
 
+good_file = """
+package main
+import os
+func main() {
+\tf, _ := os.Open("foo")
+}"""
```
Comment (Makman2) [inline tests/go/GoImportsBearTest.py]: -->
```python
good_file = """package main

import "os"

func main() {
\tf, _ := os.Open("foo")
}"""
```

File: tests/go/GoImportsBearTest.py [hunk 4]
```diff
@@ -2,9 +2,21 @@
...
+
+bad_file = """
+package main
+func main() {
+\tf, _ := os.Open("foo")
+}
+"""
```
Comment (Makman2) [inline tests/go/GoImportsBearTest.py]: -->
```python
bad_file = """package main


func main() {
\tf, _ := os.Open("foo")
}"""
```

**Comment by Ana06** [comment]
@Makman2 

Instead of:
``` Python
def load_testdata(filename):
    path = os.path.join(os.path.dirname(__file__), "test_files", filename)
    with open(path):
        return path.read()
```

I think it should be:
``` Python
def load_testdata(filename):
    path = os.path.join(os.path.dirname(__file__),
                        "test_files",
                        filename)
    with open(path, 'r') as f:
        return f.read()
```

I am not sure if I should do `f.closed` or it is not necessary.

**Comment by Makman2** [comment]
oh indeed :D
but you can omit the `'r'` flag for `open` as it's the default :)

and no you don't need to manually close the file, that's what the `with` statement takes over for you :) if you haven't heard of context managers yet:

if you use `with` on a context manager object (as `open` returns), on start of the `with` block `cm.__enter__` is called, and when you leave the `with` block, `cm.__exit__` gets called (where `cm` is the context manager object you address after `with`.

basically the context manager implementation for `open` is:
```python
def __exit__(self):
    self.close()
```

**Comment by Makman2** [comment]
alright it seems the julia-bear issue in the CI was fixed on master, so you need to do a rebase :)

> Add the original coala repository as a remote:
>
>```
>git remote add upstream https://github.com/coala-analyzer/coala.git
>```
>
>Then go to your branch and rebase it onto our master:
>
>```
>git checkout <yourbranch>
>git fetch upstream
>git rebase --interactive upstream/master
>```
>
>Now push it up. You will need to do a force push because you change history here (do not do this on >master branches :)):
>
>```
>git push --force
>```
>
>Ping us on [our gitter channel](https://gitter.im/coala-analyzer/coala) if you need help!

**Comment by Makman2** [comment]
ah and found two little typos in the commit message :P :P

> Strings in ~~mehtod~~ method verify_local_bear() were not ~~pre-slited~~ pre-splitted.

**Comment by Adrianzatreanu** [comment]
there are some issues that were fixed on master, please rebase and its kool :D

**Comment by Ana06** [comment]
Something strange has happened while rebasing... :confused:

**Comment by Ana06** [comment]
I copied from @Makman2 message and it is not 

`https://github.com/coala-analyzer/coala.git`

It is 

`https://github.com/coala-analyzer/coala-bears.git`

:scream::scream::scream::scream:

Okey, I think I can fix it

**Comment by Makman2** [comment]
aaaah damn sorry for taking the wrong address... :3

**Comment by sils1297** [comment]
@Ana06 maybe we can help you on gitter?
